messaginginapp/src/test/java/io/customer/messaginginapp/InAppMessagingStoreTest.kt

```diff
@@ -0,0 +1,372 @@
+import org.robolectric.RobolectricTestRunner
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@RunWith(RobolectricTestRunner::class)
+class InAppMessagingStoreTest : IntegrationTest() {
+
+    private var inAppEventListener = mockk<InAppEventListener>(relaxed = true)
+
+    private val module = ModuleMessagingInApp(
+        config = MessagingInAppModuleConfig.Builder(
+            siteId = TestConstants.Keys.SITE_ID,
+            region = Region.US
+        ).setEventListener(inAppEventListener).build()
+    ).attachToSDKComponent()
+
+    private lateinit var manager: InAppMessagingManager
+
+    override fun setup(testConfig: TestConfig) {
+        super.setup(
+            testConfigurationDefault {
+                diGraph {
+                    sdk {
+                        overrideDependency<ModuleMessagingInApp>(module)
+                    }
+                }
+            }
+        )
+
+        manager = SDKComponent.inAppMessagingManager
+    }
+
+    // Helper function to set up the initial state for tests
+    private fun initializeAndSetUser() {
+        manager.dispatch(InAppMessagingAction.Initialize(siteId = String.random, dataCenter = String.random, environment = GistEnvironment.PROD))
+        manager.dispatch(InAppMessagingAction.SetUserIdentifier(String.random))
+    }
+
+    override fun teardown() {
+        manager.dispatch(InAppMessagingAction.Reset)
+        super.teardown()
+    }
+
+    @Test
+    fun givenMessagesWithDifferentPriorities_whenProcessingMessages_thenHighestPriorityMessageIsProcessedFirst() = runTest {
+        val messages = listOf(
+            Message(queueId = "1", priority = 2),
+            Message(queueId = "2", priority = 1),
+            Message(queueId = "3", priority = 3)
+        )
+
+        initializeAndSetUser()
+        manager.dispatch(InAppMessagingAction.ProcessMessageQueue(messages))
+
+        val state = manager.getCurrentState()
+        state.messagesInQueue.size shouldBeEqualTo 3
+        (state.currentMessageState as? MessageState.Processing)?.message?.queueId shouldBeEqualTo "2"
+    }
+
+    @Test
+    fun givenDuplicateMessages_whenProcessingMessages_thenDuplicatesAreRemoved() = runTest {
+        val messages = listOf(
+            Message(queueId = "1"),
+            Message(queueId = "1"),
+            Message(queueId = "2")
+        )
+
+        initializeAndSetUser()
+        manager.dispatch(InAppMessagingAction.ProcessMessageQueue(messages))
+
+        val state = manager.getCurrentState()
+        state.messagesInQueue.size shouldBeEqualTo 2
+        state.messagesInQueue.any { it.queueId == "1" } shouldBe true
+        state.messagesInQueue.any { it.queueId == "2" } shouldBe true
+    }
+
+    @Test
+    fun givenMessageWithSpecificRouteRule_whenRouteChanges_thenMessageStateUpdatesAccordingly() = runTest {
+        initializeAndSetUser()
+        val message = Message(queueId = "1", properties = mapOf("gist" to mapOf("routeRuleAndroid" to "home")))
```

Comment:
I believe `routeRuleAndroid` is supposed to be a regex? We have `pageRuleContains` and `pageRuleEquals` extensions to create regex matching values received from server. I suggest we used them instead here for more accurate results.

```suggestion
        val message = Message(queueId = "1", properties = mapOf("gist" to mapOf("routeRuleAndroid" to pageRuleEquals("home"))))
```